include/deal.II/grid/filtered_iterator.h

```diff
@@ -960,10 +960,8 @@ FilteredIterator<BaseIterator> &
 FilteredIterator<BaseIterator>::
 operator = (const FilteredIterator &fi)
 {
-  Assert ((fi.state() != IteratorState::valid) || (*predicate)(fi),
-          ExcInvalidElement(fi));
-  BaseIterator::operator = (fi);
-  return *this;
+  const BaseIterator &bi = fi;
+  return operator = (bi);
```

Comment:
`mpi/step-39.mpirun=1.debug` is already failing but I can of course create another test that is more specific to this issue.